File: assignment4/IntegerLinkedList.cpp
#include "IntegerNode.cpp"
#include <cstddef>

using namespace stddef;

class IntegerLinkedList {
    IntegerNode head;
    int numberOfItems;

  public:
    IntegerLinkedList() {head = NULL; numberOfItems = 0;}
    int getNumberOfItems() {return numberOfItems;}
    void insert(int value);
    bool remove(int value);

  };


bool IntegerLinkedList::remove(int value) {
    IntegerNode current = head;
    IntegerNode previous = NULL;
    while (current != NULL && current.getData() != value) {
        previous = current;
        current = current.getNext();
    }

    if (current==NULL) //not found
        return false;
    else {
        if (current == head)
            head = head.getNext();
        else
            previous.setNext(current.getNext());
    --numberOfItems;
    return true;
}

void IntegerLinkedList::insert(int value) {
    IntegerNode nd(value);
    nd.setNext(head);
    head = nd; 
    ++numberOfItems;
    }


int main() {
    IntegerLinkedList numbers = IntegerLinkedList;
}

File: assignment4/IntegerLinkedList.java
/* The IntegerLinkedList class
   Anderson, Franceschi
*/

public class IntegerLinkedList
{
  private IntegerNode head;
  private int numberOfItems;

  /** default constructor
  *   constructors an empty list
  */
  public IntegerLinkedList( )
  {
    head = null;
    numberOfItems = 0;
  }

  /** accessor for numberOfItems
  *   @return   numberOfItmes
  */
  public int getNumberOfItems( )
  {
    return numberOfItems;
  }

  /** insert method
  *   @param    value  data to  insert
  *    inserts node at head
  */
  public void insert( int value )
  {
    IntegerNode nd = new IntegerNode( value );
    nd.setNext( head );
    head = nd;
    numberOfItems++;
  }

  /** remove method
  *   @param    value   the value to remove
  *   @return   true if value was removed, false otherwise
  */
  public boolean remove( int value )
  {
    IntegerNode current = head;
    IntegerNode previous = null;
    while ( current != null
            && current.getData( ) != value )
    {
      previous = current;
      current = current.getNext( );
    }

    if ( current == null ) // not found
      return false;
    else
    {
      if ( current == head )
        head = head.getNext( );  // remove head
      else
        previous.setNext( current.getNext( ) );

      numberOfItems--;
      return  true;
    }
  }

  /** toString
  *   @return    values in list separated by a space
  */
  public String toString( )
  {
    String listString = "";
    IntegerNode current = head;
    for ( int i = 0; i < numberOfItems; i++ )
    {
      listString += current.getData( ) + "  ";
      current = current.getNext( );
    }
    return listString;
  }
}
